kernel/src/hil/eic.rs

```diff
@@ -0,0 +1,55 @@
+//! Interface for external interrupt controller.
+
+/// Enum for selecting which edge to trigger interrupts on.
+#[derive(Debug)]
+pub enum InterruptMode {
+    RisingEdge,
+    FallingEdge,
+    HighLevel,
+    LowLevel,
+}
+
+/// Enum for enabling/disabling filter
+pub enum FilterMode {
+    FilterEnable,
+    FilterDisable,
+}
+
+/// Enum for selecting syn/asyn mode
+pub enum SynchronizationMode {
+    Synchronous,
+    Asynchronous,
+}
+
+/// Interface for EIC.
+pub trait ExternalInterruptController {
+    /// Enables EIC module
+    fn enable(&self);
+
+    /// Disables EIC module
+    fn disable(&self);
+
+    /// Enables external interrupt on line_num
+    /// In asychronous mode, all edge interrupts will be interpreted as level interrupts and the filter is disabled.
+    fn line_enable(&self, line_num: usize);
```

Comment:
Adding a type to this trait and making `line_num` of that type (see the ADC and analog comparator HILs for examples) would be more consistent.

Reply:
Yes, it eliminate to check value of `line_num` too for example if you define an enum or something with only the possible variants :)

Reply:
Passing comment: please also update this to `line_number`, we err on the side of verbosity and clarity in HIL interfaces